### ISSUE TITLE
Implement Centralized Content for VBA

### ISSUE THREAD

**Issue by xiongjaneg** [department-of-veterans-affairs/va.gov-cms#14699]
## Description
Drupal implementation for VBA Centralized Content is built based on spec #13799. See comments in that ticket for screenshots. There is no spreadsheet version of this spec, because it uses existing fields & screenshots suffice.

## Acceptance Criteria
- [x] Fields, machine names, help text, required, cardinality, character limits and any other Drupal implementation specifics are implemented as specified in the spec
- [x] Fields appear within fieldsets and in order as shown in the spec
- [x] Follow up tickets are created to track any remaining work that arises during implementation
- [x] After deploy to production, centralized content fields (of type Entity Field Fetch) are populated on VBA Facility pages through Bulk Edit (Modify field values).

**Comment by davidmpickett** [comment]
Question for @swirtSJW and @omahane:

- Once the VBA Centralized Content type exists, there would likely then be additional work to update the [VBA Facility content type](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14701) to show gray boxes of Centralized Content. Would you consider that a separate ticket, or part and parcel of this ticket?
- This ticket has some relevant examples of the gray boxes: https://github.com/department-of-veterans-affairs/va.gov-cms/issues/4055

**Comment by davidmpickett** [comment]
One other thing I want to call out that might be a stretch is that the requested model for spotlights is slightly more complex than Vet Center (I think). Rather than 1 national + up to 2 local for a total of 1-3, they want to ensure all three slots are always use, by having 2 national backfills for when no local content is there. This might be worth breaking off as a bonus enhancement for fast-follow, since it would only directly affect National Editors.

![265146980-b3263a43-127e-4608-8632-d7818d33d431](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/78b6b9bd-9337-47eb-a0d8-83ebff402856)

**Comment by xiongjaneg** [comment]
Needs the text for the documenter paragraph and the field. Sync with Dave P. on this.

**Comment by davidmpickett** [comment]
Christian's work is blocked by my review of 
- Titles of descriptors (because these will be used as basis for machine names of entity field fetch fields on VBA facility content type)
- Field types of all content input fields - make sure they are correct

**Comment by davidmpickett** [comment]
> Once the VBA Centralized Content type exists, there would likely then be additional work to update the https://github.com/department-of-veterans-affairs/va.gov-cms/issues/14701 to show gray boxes of Centralized Content. Would you consider that a separate ticket, or part and parcel of this ticket?

Per my chat with @omahane - the answer to this question is yes. Adding `entity field fetch fields` to the VBA Facility content type is the bulk of the work from an engineering perspective

**Comment by davidmpickett** [comment]
Copy and pasting the current content of the node here so I can recommend changes

- **Title:** National VBA facilities content
- **Product:** VBA facilities
- **Applied to:** [ null ]
- **Purpose:** This is used on all VBA facility pages
- Centralized content 1
  - **Content descriptor title:** VBA Facility overview
  - **Content block type:** Rich text
- Centralized content 2
  - **Content descriptor title:** Benefits hotline
  - **Content block type:** Phone number
- Centralized content 3
  - **Content descriptor title:** Hours advisory
  - **Content block type:** Rich text
- Centralized content 4
  - **Content descriptor title:** The featured "spotlight" content on all Vet Center pages
  - **Content block type:** Featured content
- Centralized content 5
  - **Content descriptor title:** Other services advisory
  - **Content block type:** Featured content
- Centralized content 6
  - **Content descriptor title:** Updates links
  - **Content block type:** List of links

**Comment by davidmpickett** [comment]
**Names are hard note:**
The field with machine name `content_block` does not map directly to either of the Drupal entities that start with the word content or block (content type / block layout). In fact it is a field that references Paragraph types

Paragraph types

**Comment by davidmpickett** [comment]
Screenshot of current state

![screencapture-staging-cms-va-gov-centralized-content-national-vba-facilities-content-2023-09-29-14_45_45](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/9879bf65-b274-4f25-adfd-208ff6586a55)

**Comment by davidmpickett** [comment]
Side question for @omahane @swirtSJW about Centralized Content that could spin off into it's own ticket:

Do the Product and Applied to fields actually do anything? They seem like vestigial fields that were intended to help with documentation, and information that previously would have been populated there could now live in Content Model Documentation instead?

![Screenshot 2023-09-29 145140](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/51967950/89f96a9f-e4f5-4e14-86d5-d88f797e3459)

**Comment by omahane** [comment]
@davidmpickett I'm not sure if they have any effect. I think they're just self-documenting.

**Comment by davidmpickett** [comment]
@omahane - proposed changes in bold. Notes in italics

- Title: **National VA benefit office content**
  - _Rationale: We do not know if this content will also be used on the VBA facility types that are outside of MVP scope. This content only applies to VA Regional Benefit Offices and VA Regional Benefit Satellite Offices. VBA facility may be too broad of a name in this instance. "Benefit Office" encompasses the two types we are designing for and excludes the others._
- Product: VBA facilities
- Applied to: [ null ]
- Purpose: **This content displays used on all regional & satellite benefit office pages.**
- Centralized content 1
  - Content descriptor title: **Benefit office overview**
  - content_block: **rich_text_char_limit_1000**
- Centralized content 2
  - Content descriptor title: **VA benefits hotline**
  - content_block: Phone number
- Centralized content 3
  - Content descriptor title: **Benefit office hours may vary**
  - content_block: **rich_text_char_limit_1000**
- Centralized content 4
  - Content descriptor title: **The featured "spotlight" content on all VA benefit office pages**
  - content_block: Featured content
    - _Question: Are you thinking that this will be a paragraph with cardinality 3 to account for the national backfill? Or do we need three separate fields here?_
- Centralized content 5
  - Content descriptor title: **Can't find the benefit service you're looking for?**
  - content_block: Featured content
- Centralized content 6
  - Content descriptor title: **Get updates from VBA**
  - content_block: List of links

**Comment by omahane** [comment]
> Question: Are you thinking that this will be a paragraph with cardinality 3 to account for the national backfill? Or do we need three separate fields here?

I don't see any way to set the cardinality of a paragraph distinct from the cardinality of the field that references it. I think it makes sense to have three of these and they get filled in by national editors and are used as 1, 2 or all 3, depending on the number of facility-specific spotlight items.

**Comment by swirtSJW** [comment]
> Side question for @omahane @swirtSJW about Centralized Content that could spin off into it's own ticket:
> 
> Do the Product and Applied to fields actually do anything? They seem like vestigial fields that were intended to help with documentation, and information that previously would have been populated there could now live in Content Model Documentation instead?
> 
@davidmpickett and @omahane 

They do not currently apply to anything in a systemic/automated way.  They are locked down fields for editors and were intended to give as much context as possible to the what and where.
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/5752113/59f8fcc9-9eab-4f42-ae00-7ca2505ba9f5)
Notice on the left, an assigned editor does not see them as fields.  Only admins see the field.  They are meant to be set up at the beginning by the admin who instantiates the Centralized content node so that we do not have all these Centralized content nodes created that nobody understands the context for.

It is similar but different than a Content Model Document.  

The product field does not change the behavior of the node itself but it does tie it to the product taxononmy and at some point can help surface these at a product level. 

It is easy now while we only have 8 Centralized content nodes to recognise what goes where, but this was built with there potentially being hundreds.

**Comment by jilladams** [comment]
@omahane Closing based on demo.
